Prevent gpg signing of commits (fixes #30)

tkw1536 opened the pull request:
When using git 2.x and the `commit.gpgsign` option is enabled, each commit is automatically signed with gpg. As noted in #30 this usually means that the user has to enter a password, or unlock the gpg key using some other method. This is a problem when having to leave the building because of fire.

This commit solves the problem by adding the --no-gpg-sign flag to the `git commit` invocation if the git version being used starts with a "2". This is achieved by checking the output of `git version` and using grep to match the version number.

tkw1536 committed:
Prevent gpg signing of commits (fixes #30)

When using git 2.x and the `commit.gpgsign` option is enabled, each
commit is automatically signed with gpg. As noted by @johnp this usally
means that the user has to enter a password, or unlock the gpg key using
some other method. This is a problem when having to leave the building
because of fire.

This commit solves the problem by adding the --no-gpg-sign flag to the
`git commit` invocation if the git version being used starts with a "2".
This is achieved by checking the output of `git version` and using grep
to match the version number.

tkw1536 commented:
/cc @johnp this implements your suggestion

glensc commented on git-fire:
```diff
@@ -42,7 +42,14 @@ fire() {
 
-	git commit -m "$message" --no-verify
+	# for git >= 2.x add the --no-gpg-sign flag
+	if (git version | grep \\s2\.) then
```
subshell is not necessary, but need `;` before `then` then.

tkw1536 committed:
Improved checking for git version

When checking that the git version is 2.x, do not use a subshell and
redirect output to /dev/null.

tkw1536 commented:
I have updated the pull request after a comment from @glensc. 
I am no longer using a sub-shell to check the git version, and am also silencing output from the version check.

tkw1536 commented:
Since nothing happened in such a long time, I'm closing this PR.
